stl/inc/xutility

```diff
@@ -4198,7 +4198,368 @@ _NODISCARD _CONSTEXPR17 move_iterator<_Iter> make_move_iterator(_Iter _It) { //
+            if (_Mystate == _State::_Holds_iter) {
+                _Myiter.~_Iter();
+            }
+            if constexpr (!is_nothrow_copy_constructible_v<const _Se&>) {
+                _Mystate = _State::_Nothing;
+            }
+            _Construct_in_place(_Mysentinel, _Right._Mysentinel);
+            _Mystate = _State::_Holds_sentinel;
+        }
+        return *this;
+    }
+
+    // clang-format off
+    template <class _Iter2, class _Se2>
+        requires convertible_to<const _Iter2&, _Iter> && convertible_to<const _Se2&, _Se>
+              && assignable_from<_Iter&, const _Iter2&> && assignable_from<_Se&, const _Se2&>
+    common_iterator& operator=(const common_iterator<_Iter2, _Se2>& _Right) noexcept(
+        is_nothrow_assignable_v<_Iter, const _Iter2&>&& is_nothrow_assignable_v<_Se, const _Se2&>&&
+            is_nothrow_constructible_v<_Iter, const _Iter2&>&&
+                is_nothrow_constructible_v<_Se, const _Se2&>) // strengthened
+    // clang-format on
+    {
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(_Right._Mystate != _State::_Nothing, "common_iterator can only be assigned if it holds a value");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+        if (_Mystate == _State::_Holds_iter && _Right._Mystate == _State::_Holds_iter) {
+            if constexpr (!is_nothrow_assignable_v<_Iter, const _Iter2&>) {
+                _Mystate = _State::_Nothing;
+            }
+            _Myiter = _Right._Myiter;
+            if constexpr (!is_nothrow_assignable_v<_Iter, const _Iter2&>) {
+                _Mystate = _State::_Holds_iter;
+            }
+        } else if (_Mystate == _State::_Holds_sentinel && _Right._Mystate == _State::_Holds_sentinel) {
+            if constexpr (!is_nothrow_assignable_v<_Se, const _Se2&>) {
+                _Mystate = _State::_Nothing;
+            }
+            _Mysentinel = _Right._Mysentinel;
+            if constexpr (!is_nothrow_assignable_v<_Se, const _Se2&>) {
+                _Mystate = _State::_Holds_sentinel;
+            }
+        } else if (_Right._Mystate == _State::_Holds_iter) {
+            if (_Mystate == _State::_Holds_sentinel) {
+                _Mysentinel.~_Se();
+            }
+            if constexpr (!is_nothrow_constructible_v<_Iter, const _Iter2&>) {
+                _Mystate = _State::_Nothing;
+            }
+            _Construct_in_place(_Myiter, _Right._Myiter);
+            _Mystate = _State::_Holds_iter;
+        } else if (_Right._Mystate == _State::_Holds_sentinel) {
+            if (_Mystate == _State::_Holds_iter) {
+                _Myiter.~_Iter();
+            }
+            if constexpr (!is_nothrow_constructible_v<_Se, const _Se2&>) {
+                _Mystate = _State::_Nothing;
+            }
+            _Construct_in_place(_Mysentinel, _Right._Mysentinel);
+            _Mystate = _State::_Holds_sentinel;
+        }
+        return *this;
+    }
+
+    ~common_iterator() noexcept() // strengthened

```

Comment:
```suggestion
    ~common_iterator()

```

Reply:
Citing @StephanTLavavej "I think this should be unconditionally noexcept"

Reply:
Ah i derped again and left `noexcept()` instead of `noexcept`.

Maybe I should refrain from pushing "trivial" fixes from my linux machine